tst/com/amazon/corretto/crypto/provider/test/EdDSATest.java

```diff
@@ -0,0 +1,284 @@
+    // Generate keys with ACCP and use BC KeyFactory to get equivalent JCE Keys
+    final byte[] message = new byte[] {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    final Signature nativeSig = Signature.getInstance("Ed25519", NATIVE_PROVIDER);
+    final Signature bcSig = Signature.getInstance("Ed25519", BOUNCYCASTLE_PROVIDER);
+    final KeyPair keyPair = nativeGen.generateKeyPair();
+
+    final PKCS8EncodedKeySpec privateKeyPkcs8 =
+        new PKCS8EncodedKeySpec(keyPair.getPrivate().getEncoded());
+    final X509EncodedKeySpec publicKeyX509 =
+        new X509EncodedKeySpec(keyPair.getPublic().getEncoded());
+
+    final KeyFactory kf = KeyFactory.getInstance("Ed25519", BOUNCYCASTLE_PROVIDER);
+
+    final PrivateKey privateKey = kf.generatePrivate(privateKeyPkcs8);
+    final PublicKey publicKey = kf.generatePublic(publicKeyX509);
+
+    // Sign with ACCP, Verify with BouncyCastle
+    nativeSig.initSign(keyPair.getPrivate());
+    nativeSig.update(message, 0, message.length);
+    byte[] signature = nativeSig.sign();
+    bcSig.initVerify(publicKey);
+    bcSig.update(message);
+    assertTrue(bcSig.verify(signature), "Native->BC: Ed25519");
+
+    bcSig.initSign(privateKey);
+    bcSig.update(message, 0, message.length);
+    signature = bcSig.sign();
+    nativeSig.initVerify(keyPair.getPublic());
+    nativeSig.update(message);
+    assertTrue(nativeSig.verify(signature), "BC->Native: Ed25519");
+  }
+
+  @Test
+  public void bcKeyValidation() throws GeneralSecurityException {
+    // Generate keys with ACCP and use BC KeyFactory to get equivalent JCE Keys
+    final KeyPair kp = nativeGen.generateKeyPair();
+    final byte[] pkACCP = kp.getPrivate().getEncoded();
+    final byte[] pbkACCP = kp.getPublic().getEncoded();
+
+    final PKCS8EncodedKeySpec privateKeyPkcs8 = new PKCS8EncodedKeySpec(pkACCP);
+    final X509EncodedKeySpec publicKeyX509 = new X509EncodedKeySpec(pbkACCP);
+
+    final KeyFactory kf = KeyFactory.getInstance("Ed25519", BOUNCYCASTLE_PROVIDER);
+
+    final byte[] pkBC = kf.generatePrivate(privateKeyPkcs8).getEncoded();
+    final byte[] pbkBC = kf.generatePublic(publicKeyX509).getEncoded();
+
+    // Confirm that ACCP & BC keys are equivalent
+    assertTrue(Arrays.equals(pkACCP, pkBC));
+    assertTrue(Arrays.equals(pbkACCP, pbkBC));
+  }
+
+  @Test
+  public void eddsaValidation() throws GeneralSecurityException {
+    // Generate keys, sign, & verify with ACCP
+    final byte[] message = new byte[] {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    final Signature eddsa = Signature.getInstance("Ed25519", NATIVE_PROVIDER);
+    final KeyPair keyPair = nativeGen.generateKeyPair();
+
+    eddsa.initSign(keyPair.getPrivate());
+    eddsa.update(message, 0, message.length);
+    final byte[] signature = eddsa.sign();
+
+    eddsa.initVerify(keyPair.getPublic());
+    eddsa.update(message, 0, message.length);
+    assertTrue(eddsa.verify(signature));
+  }
+
+  @Test
+  public void mismatchSignature() throws GeneralSecurityException {
```

Comment:
Please add some compatibility tests to ensure that ACCP behaves the same as SunJCE when the signature verification fails. For example, they both throw exceptions or they both return false.

Reply:
Resolved in latest commit.